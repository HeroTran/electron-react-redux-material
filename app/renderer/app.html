<!DOCTYPE html>
<html lang="en">
  <head>
    <title>herotran210_desktop_app</title>
    <meta name="description" content="herotran210_desktop_app">
    <meta name="keywords" content="herotran210_desktop_app">
    <meta name="author" content="herotran210_desktop_app">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32" href="../../resources/favicon.ico" />
    <link rel="icon" type="image/png" sizes="16x16" href="../../resources/favicon.ico" />
    <link rel="shortcut icon"  href="../../resources/favicon.ico" />
    <script>
      (() => {
        if (
          typeof process !== 'object' ||
          (typeof process === 'object' && !process.env.START_HOT)
        ) {
          const link = document.createElement('link');
          link.rel = 'stylesheet';
          link.href = '../dist/style.css?v' + Math.random();
          document.getElementsByTagName('head')[0].appendChild(link);
        }
      })();
    </script>
  </head>
  <body>
    <div id="root"></div>
    <script>
      if (typeof process === 'object') {
        const scripts = [];

        if (process.env.NODE_ENV === 'development') {
          // Dynamically insert the DLL script in development env in the
          // renderer process
          scripts.push('../dll/renderer.dev.dll.js');
        }
        if (process.env.START_HOT) {
          // Dynamically insert the bundled app script in the renderer process
          const port = process.env.PORT || 3001;
          scripts.push(`http://localhost:${port}/dist/renderer.dev.js`);
        } else {
          scripts.push('../dist/renderer.prod.js?v' + Math.random());
        }

        if (scripts.length) {
          document.write(
            scripts
              .map(script => `<script defer src="${script}"><\/script>`)
              .join('')
          );
        }
      }
    </script>
  </body>
</html>
