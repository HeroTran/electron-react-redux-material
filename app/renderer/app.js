import * as React from 'react';
import { HashRouter, Switch, Route } from 'react-router-dom';
import PrivateRoute from './components/Common/PrivateRoute/PrivateRoute';
import { LoginContainer, AppContainer } from './containers';
import { ROUTE_PAGE } from './functionals/Common/constants';

const App = () => {
  return (
    <HashRouter>
      <Switch>
        <Route path={ROUTE_PAGE.ROUTE_LOGIN} component={LoginContainer} />
        <PrivateRoute path={ROUTE_PAGE.ROUTE_HOME} component={AppContainer} />
      </Switch>
    </HashRouter>
  );
};
export default App;
