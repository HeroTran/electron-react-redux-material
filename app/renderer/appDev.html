<!DOCTYPE html>
<html lang="en">
  <head>
    <title>herotran210_desktop_app</title>
    <meta charset="UTF-8">
    <meta name="description" content="herotran210_desktop_app">
    <meta name="keywords" content="herotran210_desktop_app">
    <meta name="author" content="herotran210_desktop_app">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32" href="../../resources/favicon.ico" />
    <link rel="icon" type="image/png" sizes="16x16" href="../../resources/favicon.ico" />
    <link rel="shortcut icon"  href="../../resources/favicon.ico" />
  </head>
  <body>
    <div id="root"></div>
    <script>
      if (typeof process === 'object') {
        const scripts = [];
        const port = process.env.PORT || 3001;
        scripts.push('../dll/renderer.dev.dll.js?v' + Math.random());
        scripts.push(
          `http://localhost:${port}/dist/renderer.dev.js?v` + Math.random()
        );

        if (scripts.length) {
          document.write(
            scripts
              .map((script) => `<script defer src="${script}"><\/script>`)
              .join('')
          );
        }
      }
    </script>
  </body>
</html>
