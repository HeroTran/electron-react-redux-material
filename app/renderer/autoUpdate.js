const { ipcRenderer } = require('electron');

const select = (selector) => document.querySelector(selector);

const container = select('#messages');
const progressBar = select('#progressBar');
const version = select('#version');

ipcRenderer.on('message', (event, text) => {
  const message = document.createElement('div');
  message.innerHTML = text;
  container.appendChild(message);
});

ipcRenderer.send('app_version');
ipcRenderer.on('app_version', (event, arg) => {
  ipcRenderer.removeAllListeners('app_version');
  version.innerText = `Version ${arg.version}`;
});

ipcRenderer.on('download-progress', (event, text) => {
  progressBar.style.width = `${text}%`;
});

ipcRenderer.on('update-downloaded', () => {
  ipcRenderer.send('restart_app');
});
