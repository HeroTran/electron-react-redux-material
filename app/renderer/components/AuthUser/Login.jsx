import React, { useState, useEffect, useCallback } from 'react';
import { Redirect } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import Grid from '@material-ui/core/Grid';
import { ROLE, ROUTE_PAGE } from '../../functionals/Common/constants';
import { STATUS_AUTH } from '../../functionals/AuthUser/constants';
import { openLink } from '../../utils/electronHelper';
import { isCheckAuth } from '../../utils/localStoreHelper';
import i18n from '../../i18n';
import './login.scss';

const Login = (props) => {
  const { isLoading, currentUser, errorAuth, handleLogin, history } = props;
  const [hasError, setHasError] = useState('');
  const [openEye, setOpenEye] = useState(true);
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = (data) => {
    setHasError('');
    handleLogin(data.email, data.password);
  };
  useEffect(() => {
    if (errorAuth) {
      const { status } = errorAuth.data;
      if (status === STATUS_AUTH.ACCOUNT_NOT_ACTIVE) {
        setHasError(i18n.t('validation.accountNotActive'));
      } else if (
        status === STATUS_AUTH.INCORRECT_PASSWORD ||
        status === STATUS_AUTH.ACCOUNT_NOT_EXIST
      ) {
        setHasError(i18n.t('validation.userPassIncorrect'));
      }
    } else if (currentUser) {
      if (currentUser.role === ROLE.ROLE_TEACHER) {
        history.push(ROUTE_PAGE.ROUTE_HOME);
      } else {
        setHasError(i18n.t('validation.accountNotActive'));
      }
    }
  }, [currentUser, errorAuth]);

  useEffect(() => {
    setHasError('');
  }, []);

  const handleClickEye = useCallback(() => {
    setOpenEye(!openEye);
  }, [openEye]);

  const handleClickOpenForgotPassword = useCallback((event) => {
    event.preventDefault();
    openLink(ROUTE_PAGE.ROUTE_FORGOT_PASSWORD);
  }, []);
  if (isCheckAuth()) {
    return <Redirect to={ROUTE_PAGE.ROUTE_HOME} />;
  }
  return (
    <div className="wrapper-login">
      <Grid container spacing={0}>
        <Grid item xs={12} sm={6}>
          <div className="form-background">
            <div className="bg-login" />
          </div>
        </Grid>
        <Grid item xs={12} sm={6}>
          <h2>Update Login version 0.0.18</h2>
          <div className="form-login">
            <div className="main-login">
              <form onSubmit={handleSubmit(onSubmit)}>
                <div className="group-form">
                  <div className="each-form">
                    <input
                      className="form-control input-form"
                      name="email"
                      placeholder={`${i18n.t('login.email')}`}
                      ref={register({
                        required: `${i18n.t('login.email')} ${i18n.t(
                          'validation.required'
                        )}`,
                        pattern: {
                          value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                          message: i18n.t('validation.formatEmail'),
                        },
                      })}
                    />
                    <i className="icon icon-email" />
                    {errors.email && (
                      <span className="error">{errors.email.message}</span>
                    )}
                  </div>
                  <div className="each-form">
                    <input
                      className="form-control input-form"
                      placeholder={`${i18n.t('login.password')}`}
                      type={openEye ? `password` : `text`}
                      name="password"
                      ref={register({ required: true })}
                    />
                    <i className="icon icon-password" />
                    <i
                      onClick={() => handleClickEye(false)}
                      className={`icon ${
                        openEye ? `icon-eye-open` : `icon-eye-close`
                      }`}
                    />
                    {errors.password && (
                      <span className="error">
                        {`${i18n.t('login.password')} ${i18n.t(
                          'validation.required'
                        )}`}
                      </span>
                    )}
                  </div>
                  <div className="each-form each-form-center">
                    <a
                      onClick={(e) => handleClickOpenForgotPassword(e)}
                      className="link"
                    >
                      {i18n.t('login.forgotPassword')}
                    </a>
                  </div>
                  <div className="each-form each-form-center">
                    <button
                      className={
                        isLoading
                          ? `btn btn-primary btn-login`
                          : `btn btn-primary`
                      }
                      type="submit"
                      name="signin"
                      disabled={isLoading}
                    >
                      {i18n.t('login.signIn')}
                      {isLoading ? <i className="loading" /> : ''}
                    </button>
                  </div>
                </div>
                {hasError && hasError.length > 0 && (
                  <div className="form-error">
                    <p className="error">{hasError}</p>
                  </div>
                )}
              </form>
            </div>
          </div>
        </Grid>
      </Grid>
    </div>
  );
};
export default Login;
