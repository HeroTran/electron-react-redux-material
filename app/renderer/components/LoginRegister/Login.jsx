import * as React from 'react';
import { withRouter } from 'react-router-dom';
import './login.scss';

const Login = (props) => {
  return (
    <div>
      <button type="button" onClick={props.history.goBack}>
        Back
      </button>
      <h1>Login page</h1>
    </div>
  );
};
export default withRouter(Login);
