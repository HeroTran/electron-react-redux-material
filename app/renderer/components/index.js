import App from './App/App';
import Login from './AuthUser/Login';
import ClassList from './ClassList/ClassList';
import Home from './Home/Home';
import NoMatch from './Common/NoMatch/NoMatch';
import PrivateRoute from './Common/PrivateRoute/PrivateRoute';
import OpenSlot from './OpenSlot/OpenSlot';
import Profile from './Profile/Profile';
import Setting from './Setting/Setting';
import VirtualClassRoom from './VirtualClassRoom/VirtualClassRoom';

export {
  App,
  Login,
  ClassList,
  NoMatch,
  PrivateRoute,
  Home,
  OpenSlot,
  Profile,
  Setting,
  VirtualClassRoom,
};
