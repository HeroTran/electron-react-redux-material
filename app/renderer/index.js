import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { CssBaseline } from '@material-ui/core';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { I18nextProvider } from 'react-i18next';
import configureStore, { history } from './functionals/store';
import App from './app';
import i18n from './i18n';
import './styles/styles.scss';

export const store = configureStore();
const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
});
const render = () => {
  ReactDOM.render(
    <Provider store={store}>
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <ConnectedRouter history={history}>
          <I18nextProvider i18n={i18n}>
            <App />
          </I18nextProvider>
        </ConnectedRouter>
      </MuiThemeProvider>
    </Provider>,
    document.getElementById('root') || document.createElement('div') // for testing purposes
  );
};
render();
if (module.hot) {
  module.hot.accept(['./i18n', './components/App/App'], () => {
    ReactDOM.unmountComponentAtNode(document.getElementById('root'));
    render();
  });
}
