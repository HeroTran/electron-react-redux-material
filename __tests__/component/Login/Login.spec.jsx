import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../../app/renderer/components/AuthUser/Login';

describe('should be rendered without error', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
