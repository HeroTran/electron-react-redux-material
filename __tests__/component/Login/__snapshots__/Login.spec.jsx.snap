// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be rendered without error matches snapshot 1`] = `
<div
  className="wrapper-login"
>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
    >
      <div
        className="form-background"
      >
        <div
          className="bg-login"
        />
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
    >
      <div
        className="form-login"
      >
        <div
          className="main-login"
        >
          <div
            className="logo-login"
          />
          <form
            onSubmit={[Function]}
          >
            <div
              className="group-form"
            >
              <div
                className="each-form"
              >
                <input
                  className="form-control input-form"
                  name="email"
                  placeholder="Email"
                />
                <i
                  className="icon icon-email"
                />
              </div>
              <div
                className="each-form"
              >
                <input
                  className="form-control input-form"
                  name="password"
                  placeholder="Password"
                  type="password"
                />
                <i
                  className="icon icon-password"
                />
                <i
                  className="icon icon-eye-open"
                  onClick={[Function]}
                />
              </div>
              <div
                className="each-form each-form-center"
              >
                <a
                  className="link"
                  onClick={[Function]}
                >
                  Forgot password?
                </a>
              </div>
              <div
                className="each-form each-form-center"
              >
                <button
                  className="btn btn-primary"
                  name="signin"
                  type="submit"
                >
                  Sign in
                  
                </button>
              </div>
            </div>
            
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
